### PR TITLE
Session storage bug fixes

### PR DESCRIPTION
**Pull request by kamilszostak**
Fix for #217.

### CODE REVIEW

**Comment by dnduffy** [comment]
:shipit:

File: JavaScript/JavaScriptSDK/AppInsights.ts [hunk 1]
```diff
@@ -78,7 +78,16 @@ module Microsoft.ApplicationInsights {
                 enableSessionStorageBuffer: () => this.config.enableSessionStorageBuffer
             }
 
-            this.context = new ApplicationInsights.TelemetryContext(configGetters);
+            // enable session storage buffer experiment		
+            var enableExperiment = new SplitTest().isEnabled(this.config.instrumentationKey, 5);
+            this.config.enableSessionStorageBuffer = enableExperiment;
+
+            this.context = new ApplicationInsights.TelemetryContext(configGetters); this.context = new ApplicationInsights.TelemetryContext(configGetters);
```
Comment (Dmitry-Matveev) [inline JavaScript/JavaScriptSDK/AppInsights.ts]: this.context is duplicated on this line.

Comment (kamilszostak) [inline JavaScript/JavaScriptSDK/AppInsights.ts]: Good catch! :+1:

**Comment by Dmitry-Matveev** [comment]
:shipit:

File: JavaScript/JavaScriptSDK/AppInsights.ts [hunk 2]
```diff
@@ -61,6 +61,11 @@ module Microsoft.ApplicationInsights {
                 }
             }
 
+            var tab = ["a", "b"];
```
Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK/AppInsights.ts]: what is that?

File: JavaScript/JavaScriptSDK/SendBuffer.ts [hunk 3]
```diff
@@ -179,15 +180,23 @@ module Microsoft.ApplicationInsights {
         }
 
         private removePayloadsFromBuffer(payloads: string[], buffer: string[]): string[] {
-            var cleared: string[] = [];
+            var remaining: string[] = [];
```
Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK/SendBuffer.ts]: just wanted to make sure you didn't forget unit tests for this change (or it's covered by existing ones)

File: JavaScript/JavaScriptSDK/AppInsights.ts (hunk 2 above)
Comment (kamilszostak) [inline JavaScript/JavaScriptSDK/AppInsights.ts]: how this got here :) ? I was doing some testing and forgot to remove it.

File: JavaScript/JavaScriptSDK/SendBuffer.ts (hunk 3 above)
Comment (kamilszostak) [inline JavaScript/JavaScriptSDK/SendBuffer.ts]: Yes, it's covered in SendBuffer.tests.ts file.

**Comment by MaxShehovtsov** [comment]
:shipit:

**Pull request merged**